net.go

```diff
@@ -126,41 +170,59 @@ func dialSession(d net.Dialer, network, addr string, ctx *Ctx, flags DialFlags,
 		}
 		// TODO: use operating system default certificate chain?
 	}
+	return ctx, nil
+}
 
-	c, err := d.Dial(network, addr)
+func createConnection(context context.Context, d net.Dialer, network,
+	addr string) (net.Conn, error) {
+	c, err := d.DialContext(context, network, addr)
 	if err != nil {
 		return nil, err
 	}
-	conn, err := Client(c, ctx)
-	if err != nil {
-		c.Close()
-		return nil, err
-	}
-	if session != nil {
-		err := conn.setSession(session)
-		if err != nil {
-			c.Close()
-			return nil, err
-		}
-	}
+	return c, nil
+}
+
+func applyFlags(conn *Conn, host string, flags DialFlags) error {
```

Comment:
```suggestion
func handshake(conn *Conn, host string, flags DialFlags) error {
```

The essence of the function not about flags applying.

Reply:
Updated.